ocaml/sdk-gen/csharp/templates/HTTP_actions.mustache

```diff
@@ -21,11 +46,11 @@ namespace XenAPI
         }
 
 {{#http_actions}}
-        public static void {{name}}(HTTP.{{delegate_type}} {{delegate_name}}, HTTP.FuncBool cancellingDelegate, int timeout_ms,
-            string hostname, IWebProxy proxy, string path, {{{sdkargs_decl}}})
+        public static void {{name}}(HTTP.{{#isPut}}UpdateProgressDelegate progressDelegate{{/isPut}}{{#isGet}}DataCopiedDelegate dataCopiedDelegate{{/isGet}}, HTTP.FuncBool cancellingDelegate, int timeout_ms,
```

Comment:
If there just two possible cases, "Put" or "Get", I think 1 var "isPut" is enough, like 
```
HTTP.{{#isPut}}UpdateProgressDelegate progressDelegate{{/isPut}}{{^isPut}}DataCopiedDelegate dataCopiedDelegate{{/isPut}}
```

Reply:
At the moment, yes, we only have Put and Get methods, but I used two separate variables isPut and isGet in case it is expanded in future to include other types of methods like Post etc.